SQL_DML_HW

nina25xx opened the pull request.

nina25xx committed:
Add files via upload

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task1.sql:
```diff
@@ -0,0 +1,203 @@
+    'A computer hacker learns about the true nature of reality',
+    1999,
+    (SELECT language_id FROM language WHERE name = 'English' LIMIT 1),
+    3,  -- rental duration (weeks)
```
This should be 21 days (3 weeks) according to the task requirements. The task notes specifically mention "Duration is added in days, not in weeks"

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task1.sql:
```diff
@@ -0,0 +1,203 @@
+    'A secret agent learns to manipulate time',
+    2020,
+    (SELECT language_id FROM language WHERE name = 'English' LIMIT 1),
+    2,  -- rental duration (weeks)
```
Replace weeks for days too.

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task1.sql:
```diff
@@ -0,0 +1,203 @@
+    'A poor family infiltrates a wealthy household',
+    2019,
+    (SELECT language_id FROM language WHERE name = 'English' LIMIT 1),
+    1,  -- rental duration (weeks)
```
Replace weeks for days.

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task1.sql:
```diff
@@ -0,0 +1,203 @@
+        ('Song','Kang-ho'),
+        ('Choi','Woo-shik'),
+        ('Park','So-dam')
+)AS actor_data(first_name, last_name)
```
Good use of the VALUES clause for multiple insertions. You should add a WHERE NOT EXISTS clause to prevent duplicates if the script is run multiple times.

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task1.sql:
```diff
@@ -0,0 +1,203 @@
+          -- Parasite 
+        (266, 1041), -- Song Kang-ho
+        (267, 1041), -- Choi Woo-shik
+        (268, 1041)  -- Park So-dam
```
This violates the requirement "No hardcoded ids". You should use subqueries to get the IDs dynamically. Consider using a join between actor names and film titles.

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task1.sql:
```diff
@@ -0,0 +1,203 @@
+VALUES 
+		(1038,1),
+		(1040,1),
+		(1041,1);
```
These are hardcoded film_id values. Use subqueries to get the IDs based on film titles. Also, consider selecting a store randomly as per the task requirements.

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task1.sql:
```diff
@@ -0,0 +1,203 @@
+	first_name='Nina',
+	last_name='Petrovic',
+	email='culibrkminja@gmail.com',
+	address_id=258
```
Hardcoded address_id and customer_id. Use a subquery for the address_id to select it randomly as per requirements.

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task1.sql:
```diff
@@ -0,0 +1,203 @@
+RETURNING payment_id, amount, payment_date;
+DELETE FROM rental 
+WHERE customer_id = (SELECT customer_id FROM customer WHERE first_name = 'Nina' AND last_name = 'Petrovic')
+RETURNING rental_id, inventory_id, rental_date;
```
Good use of subqueries to avoid hardcoding the customer_id. Consider using a transaction (BEGIN/COMMIT) to ensure both operations succeed or fail together.

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task1.sql:
```diff
@@ -0,0 +1,203 @@
+FROM inventory
+WHERE film_id IN (SELECT film_id FROM film WHERE title = 'The Matrix')
+LIMIT 1
+RETURNING *;
```
The staff selection doesn't meet the requirement: "Staff_id is working at the same store where film is". You need to join with the store table to ensure this.

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task1.sql:
```diff
@@ -0,0 +1,203 @@
+        '2017-02-15'
+    FROM latest_rental lr      
+    RETURNING *;
+COMMIT;
```
The payment amount is hardcoded (4.99) instead of being calculated based on the formula in the requirements. You need to implement the calculation based on rental_duration, actual_duration, and replacement_cost.

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task2.sql:
```diff
@@ -0,0 +1,28 @@
+--1,  Create table ‘table_to_delete’ and fill it with the following query:
```
For Task 2, you need to include the actual SQL commands you executed, not just comments

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task2.sql:
```diff
@@ -0,0 +1,28 @@
+--row_estimate (numbers of rowsin table): 9,999,898
+--total_bytes(space of table in bytes):602611712 = 602.611712
+--index_bytes(size of indexs in table): 0
+--toast_bytes: 8192
```
Good observations, but include the actual SQL query you used to get these metrics.

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task2.sql:
```diff
@@ -0,0 +1,28 @@
+
+--3, Issue the following DELETE operation on ‘table_to_delete’:
+--it took 19s to delete, time: 22:53:41 till 22:54:00
+--but when i start query again, nothing changed in memory, all is the same
```
Include the actual DELETE command you executed. Also, you need to document the space consumption after the DELETE operation.

filipenkovad commented on SQL_DML_Nina_Petrovic_HW_Task2.sql:
```diff
@@ -0,0 +1,28 @@
+--total_bytes: 8192
+--toas_bytes: 8192
+--index_bytes: 0
+--table and total is also: 8192 bytes
```
Include the actual TRUNCATE command. Your observations are good, but the task requires a more structured comparison between DELETE and TRUNCATE operations

nina25xx committed:
SQL_DML_HW

Thank you for pointing out the mistakes, I tried to correct them and I hope it's better now... thank you for your help :)

And I have one question, related to second task, what exactly does this part row_estimated : -1 mean? did it come during delete part?
